src/samplers/support/hmc_helper.jl

```diff
@@ -25,7 +25,12 @@ end
 end
 
 Base.convert{N,T<:Real}(::Type{T}, d::Dual{N,T})  = d.value
-Base.convert(::Type{Float64}, d::Dual{0,Int}) = round(Int, d.value)
+Base.convert{N}(::Type{Int}, d::Dual{N,Float64}) = round(Int, d.value)
```

Comment:
Do you need to be extending the conversion methods of a different package's type here? That can change the behavior of unrelated code